perception/mil_vision/include/mil_vision_lib/image_acquisition/ros_camera_stream.hpp

```diff
@@ -292,6 +292,7 @@ operator[](int i)
   {
     auto err = "ROSCameraStream: The circular buffer index you are trying to acess is out of bounds:\n"_s + e.what();
     ROS_WARN_THROTTLE_NAMED(1, "ROSCameraStream", err.c_str());
+    _mtx.unlock();
```

Comment:
You're right. Lock guard is a better answer. This is a fix though, so I'll approve and if he wants to submit a PR replacing these in more places he can.